server/src/test/java/org/elasticsearch/indices/recovery/RecoverySourceHandlerTests.java

```diff
@@ -462,29 +462,72 @@ public void indexTranslogOperations(
         assertThat(receivedSeqNos, equalTo(sentSeqNos));
     }
 
-    private Engine.Index getIndex(String id) {
-        final LuceneDocument document = new LuceneDocument();
-        document.add(new TextField("test", "test", Field.Store.YES));
-        final Field idField = IdFieldMapper.standardIdField(id);  // TODO tsdbid field could be different.
-        final Field versionField = new NumericDocValuesField("_version", Versions.MATCH_ANY);
-        final SeqNoFieldMapper.SequenceIDFields seqID = SeqNoFieldMapper.SequenceIDFields.emptySeqID();
-        document.add(idField);
-        document.add(versionField);
-        document.add(seqID.seqNo);
-        document.add(seqID.seqNoDocValue);
-        document.add(seqID.primaryTerm);
-        final BytesReference source = new BytesArray(new byte[] { 1 });
-        final ParsedDocument doc = new ParsedDocument(
-            versionField,
-            seqID,
-            id,
-            null,
-            Arrays.asList(document),
-            source,
-            XContentType.JSON,
-            null
-        );
-        return new Engine.Index(new Term("_id", Uid.encodeId(doc.id())), randomNonNegativeLong(), doc);
+    private interface TestCase {
+        Engine.Index createIndexOp(int docIdent);
+    }
+
+    private class StandardModeTestCase implements TestCase {
+        private final MapperService mapper;
+
+        private StandardModeTestCase() throws IOException {
```

Comment:
I think you wanted to randomly choose between the two test-cases? This one is not in use now and I think it should be.

Reply:
Boo. yeah. Sorry.